### DISCUSSION TITLE
How to create a connexion pool with rocket_contrib

### DISCUSSION THREAD

**Discussion by folivi** [rwf2/Rocket#1766] [Questions] [answered]
Hi,

How can a create and share a mongodb pool inside my application?

I tried this in my main.rs file 

```
#[macro_use] extern crate rocket;
#[macro_use] extern crate rocket_contrib;

#[get("/")]
fn index() -> &'static str {
    "Hello, world!"
}

use rocket_contrib::databases::mongodb;


#[database("db")]
struct AppMongoDb(mongodb::Client);


#[rocket::main]
async fn main() {
    rocket::build()
        .mount("/hello", routes![index])
        .launch()
        .await;
}
```

and this is what I have in my cargo file:

```
[package]
name = "rocket_graphql"
version = "0.1.0"
edition = "2018"


[dependencies]
rocket = "0.5.0-rc.1"
dotenv = "0.15.0"
mongodb = "2.0.0-beta.2"
tokio = { version = "1.8.0", features = ["full"] }

[dependencies.rocket_contrib]
version = "0.4.10"
default-features = false
features = ["mongodb_pool"]
[global.databases]
db = {url = "mongodb+srv://------"}

```

I get an error on this line 
#[database("db")]
saying "Unresolved import 'rocket' no 'Outcome' in the root.

Can you please help me fix this? and if possible explain why it's happening?

Thank you very much

**Comment by SergioBenitez** [comment]
Please read the CHANGELOG for 0.5.

**Reply by folivi** [reply]
Thank you for your reply by a little guidance would be highly appreciated...
I'm very new to Rust and Rocket.
From what I saw rocket_sync_db_pools does not have support for mongodb yet.
Would you suggest to use the mongodb rust crate directly and pass it to the manage function?

**Comment by folivi** [comment, marked as answer]
I ended up doing this. 
Hope it helps anyone.

If this is not correct/safe to do so, please advise.

```
#![feature(proc_macro_hygiene, decl_macro)]
use mongodb::{
    Client, Database,
};
use rocket::{self, get, State};

#[get("/")]
async fn index(mongo_db: &State<AppDataPool>) -> &'static str {
    for coll_name in mongo_db.mongo.list_collection_names(None).await {
        println!("collection: {:?}", coll_name);
    }
    "helloooo"
}

struct AppDataPool {
    mongo: Database,
}

#[rocket::main]
async fn main() {
    let client = Client::with_uri_str("----").await;

    let db = client.unwrap().database("my-db");
    for coll_name in db.list_collection_names(None).await {
        println!("collection: {:?}", coll_name);
    }

    rocket::build()
        .manage(AppDataPool { mongo: db })
        .mount("/", rocket::routes![index])
        .launch()
        .await
        .expect("server to launch");
}

```

**Reply by jebrosen** [reply]
This is a reasonable way to use mongodb in a Rocket application today; note that `Client` has an internal connection pool already so it is not necessary to use a separate connection pooling library.

#1117 tracks more comprehensive asynchronous database connection pool support, which would make it easier to set up and to configure the database in `Rocket.toml`.
